### PR TITLE
Passing Arguments to the Executor

### PR DESCRIPTION
**Pull request by dmikey**
The old way had this being 

```json
{
    "parameters" : [{
        "name" : "foo",
        "value" : "bar"
    }]
}
```

We lost some context along the way, and this was a bug.

We should be getting.

```json
{
    "arguments" : [{
        "foo",
        "bar"
    }]
    "parameters" : [{
        "name" : "foo",
        "value" : "bar"
    }]
}
```

I need to double check the runtime if takes flags correctly.

### CODE REVIEW

**Review by xmariachi — APPROVED** [review]
LGTM

**Comment by xmariachi** [comment]
> The old way had this being
> 
> ```json
> {
>     "parameters" : [{
>         "name" : "foo",
>         "value" : "bar"
>     }]
> }
> ```
> 
> We lost some context along the way, and this was a bug.
> 
> We should be getting.
> 
> ```json
> {
>     "arguments" : [{
>         "foo",
>         "bar"
>     }]
>     "parameters" : [{
>         "name" : "foo",
>         "value" : "bar"
>     }]
> }
> ```
> 
> I need to double check the runtime if takes flags correctly.

If the new way is to have 
config/env_vars, please let's also re-edit original comment :) Thanks!

File: executor/executor.go [hunk 1]
```diff
@@ -39,6 +39,9 @@ func New(log zerolog.Logger, options ...Option) (*Executor, error) {
 	}
 	cfg.RuntimeDir = runtime
 
+	// todo: fix for windows
+	cfg.DriversRootPath = cfg.RuntimeDir + "/extensions"
```
Comment (Maelkum) [inline executor/executor.go:43]: `filepath.Join` will use the correct path separator for the OS:

```suggestion
	cfg.DriversRootPath = filepath.Join(cfg.RuntimeDir, "extensions")
```

File: executor/command.go [hunk 2]
```diff
@@ -33,10 +34,13 @@ func (e *Executor) createCmd(paths requestPaths, req execute.Request) *exec.Cmd
...
 
-	// Function arguments.
+	// Append just ordinal Function arguments first
+	args = append(args, req.Arguments...)
+
+	// Append Function parameters with values
 	for _, param := range req.Parameters {
 		if param.Value != "" {
-			args = append(args, param.Value)
+			args = append(args, param.Name + "=" + param.Value)
```
Comment (Maelkum) [inline executor/command.go:43]: What's the difference between arguments and parameters? I don't think this brings anything compared to the previous version, functionality-wise. Speaking from the POV of just _I am executing a binary and giving it some arguments_.

The same thing can be done with the previous version by using:

```json
{
	"parameters": [
		{ "value": "key1=val1" },
		{ "value": "key2=val2" 	}
	]
}
```

I don't think executables typically use `key=val` syntax for CLI flags (with no dashes or anything). Not enough that we have such first-class support for that syntax.

Plus, with the previous version we can support all variants:

a) key-value pairs
```go
	 []execute.Parameter{
		{Value: "key1=val1"},
		{Value: "key2=val2"},
		{Value: "key3=val3"},
	}
```

b) Typical UNIX flags:

```go
	 []execute.Parameter{
		{Value: "--file"},
		{Value: "/tmp/whatever.txt"},
		{Value: "--size"},
		{Value: "64MB"},
	}
```

c) Windows whatevers:

```go
	[]execute.Parameter{
		{Value: "dir"},
		{Value: "/s"},
		{Value: "/q"},
	}
```

If anything, we can just remove `Parameters` completely and just use a string array.

**Review by Maelkum — CHANGES_REQUESTED** [review]
I'm not sure if I'm missing something, but this doesn't seem to bring any new functionality?

**Pull request closed without merging**